### ISSUE TITLE
Unexpected proof success in the interactive mode

### ISSUE THREAD

**Issue by Ailrun** [Beluga-lang/Beluga#112]
When I typed the following things in the order on beli, I got `Proof complete! (No subgoals left.)` message.

```
%: load examples/stlc.bel
%: prove tpr
[|- oft M A] -> [|- eval M M'] -> [|- oft M' A]
2
intros
split z6
split z6
split z6
split z6
split z6
split z6
```

**Comment by tsani** [comment]
Just to clarify: the bug is in the totality checker, and is related to #107. The totality checker erroneously believes that many splits are impossible.

You can replicate this bug non-interactively:

```
rec tps : [ ⊢ oft M A ] → [ ⊢ eval M M' ] → [ ⊢ oft M' A ] =
fn d ⇒ fn e ⇒
case e of
  | [ |- e_c] => impossible e
  | [ |- e_lam ] => impossible e
  | [ |- e_app E1 E2] => impossible e
;
```

**Comment by tsani** [comment]
Also FWIW I'm unable to completely replicate this bug on master. I can keep on splitting, but the goals never actually end up solved. The behaviour on master is expected.

**Comment by Ailrun** [comment]
@tsani I checked it on master again, and it still works.
Maybe there are some version-related issues. The followings are results of `opam switch` and `opam list` in my machine.
```
#  switch                      compiler                    description
→  ocaml-base-compiler.4.07.1  ocaml-base-compiler.4.07.1  ocaml-base-compiler.4.07.1
```

```
# Packages matching: installed
# Name                  # Installed # Synopsis
base-bigarray           base
base-bytes              base        Bytes library distributed with the OCaml compiler
base-threads            base
base-unix               base
biniou                  1.2.1       Binary data format designed for speed, safety, ease of use and backward compatibility as protocols evolve
conf-m4                 1           Virtual package relying on m4
cppo                    1.6.6       Code preprocessor like cpp for OCaml
dune                    1.11.1      Fast, portable and opinionated build system
dune-build-info         1.11.1      Embed build informations inside executable
easy-format             1.3.2       High-level and functional interface to the Format module of the OCaml standard library
extlib                  1.7.6       A complete yet small extension for OCaml standard library (reduced, recommended)
gen                     0.5.2       Iterators for OCaml, both restartable and consumable
merlin                  3.3.2       Editor helper, provides completion, typing and source browsing in Vim and Emacs
ocaml                   4.07.1      The OCaml compiler (virtual package)
ocaml-base-compiler     4.07.1      Official release 4.07.1
ocaml-config            1           OCaml Switch Configuration
ocaml-migrate-parsetree 1.4.0       Convert OCaml parsetrees between different versions
ocamlbuild              0.14.0      OCamlbuild is a build system with builtin rules to easily build most OCaml projects.
ocamlfind               1.8.1       A library manager for OCaml
ppx_derivers            1.2.1       Shared [@@deriving] plugin registry
ppx_tools_versioned     5.2.3       A variant of ppx_tools based on ocaml-migrate-parsetree
result                  1.4         Compatibility Result module
sedlex                  2.1         An OCaml lexer generator for Unicode
uchar                   0.0.2       Compatibility library for OCaml's Uchar module
yojson                  1.7.0       Yojson is an optimized parsing and printing library for the JSON format
```

**Comment by tsani** [comment]
Strange. I'll try on OCaml 4.07.0. I was using 06. Just to clarify: variable z6 in your example is the `[|- eval M M']` judgment? Or is the typing judgment?

**Comment by tsani** [comment]
On OCaml 4.07.0 I can keep on splitting indefinitely on both the evaluation judgment and on the typing judgment.

**Comment by Ailrun** [comment]
It was the problem of the file `examples/stlc.bel`. It is fixed in 637e33c610c695c176668197d5068cae9d30f509. Thanks, @tsani!
